classwork/classwork2.2/main.cpp

```diff
@@ -0,0 +1,205 @@
+            })
+                                   | std::views::join_with(' ');
+
+            for (const auto& value : transformed_row) {
+                outputFile << value;
+            }
+            outputFile << '\n';
+        }
+        outputFile.close();
+    }
+
+    // Process image based on favorite and unfavorite colors
+    void process(const Color &favoriteColor, const Color &unfavoriteColor = Color(-1, -1, -1)) {
+        for (int i = 0; i < HEIGHT; ++i) {
+            for (int j = 0; j < WIDTH; ++j) {
+                if (pixels[i][j] == std::make_tuple(favoriteColor.getRed(), favoriteColor.getGreen(), favoriteColor.getBlue())) {
+                    if (i > 0 && j > 0) {
+                        pixels[i - 1][j - 1] = std::make_tuple(favoriteColor.getRed(), favoriteColor.getGreen(), favoriteColor.getBlue());
+                    }
+                }
+                if (pixels[i][j] == std::make_tuple(unfavoriteColor.getRed(), unfavoriteColor.getGreen(), unfavoriteColor.getBlue())) {
+                    pixels[i][j] = std::make_tuple(favoriteColor.getRed(), favoriteColor.getGreen(), favoriteColor.getBlue());
+                }
+            }
+        }
+    }
+
+private:
+    // Function to split string by a delimiter
+    std::vector<std::string> split(const std::string &str, char delimiter) {
+        std::vector<std::string> tokens;
+        std::string token;
+        std::istringstream tokenStream(str);
+        while (std::getline(tokenStream, token, delimiter)) {
+            tokens.push_back(token);
+        }
+        return tokens;
+    }
+
+    // Function to validate if a string represents a valid integer in the range [0, 255]
+    bool isValidColorValue(const std::string &str) {
+        try {
+            int value = std::stoi(str);
+            return (value >= 0 && value <= 255);
+        } catch (...) {
+            return false;
+        }
+    }
+
+    // Function to parse a line into a vector of pixels
+    std::vector<std::tuple<int, int, int>> parseLine(const std::string &line) {
+        std::vector<std::tuple<int, int, int>> pixels;
+        std::vector<std::string> parts = split(line, ' ');
+        for (const std::string &part : parts) {
+            std::vector<std::string> colorValues = split(part, ',');
+            if (colorValues.size() != 3 ||
+                !isValidColorValue(colorValues[0]) ||
+                !isValidColorValue(colorValues[1]) ||
+                !isValidColorValue(colorValues[2])) {
+                throw std::invalid_argument("Invalid pixel format");
+            }
+            pixels.emplace_back(std::stoi(colorValues[0]), std::stoi(colorValues[1]), std::stoi(colorValues[2]));
+        }
+        return pixels;
+    }
+};
+
+int main(int argc, char *argv[]) {
+    if (argc != 6 && argc != 9) {
+        std::cerr << "Usage: " << argv[0] << " <input file> <R> <G> <B> <output file> [unfavorite R] [unfavorite G] [unfavorite B]\n";
```

Comment:
minor: with modern C++ we can use 
`std::println(stderr, "Usage: {} ...", argv[0]);`
Actually println can be used with any output stream: stdout, stderr, ofstream, sstringstream and so on